apps/editing-toolkit/editing-toolkit-plugin/block-patterns/class-block-patterns.php

```diff
@@ -82,75 +97,50 @@ public function register_patterns() {
-			if ( substr( $pattern, -5 ) === '.json' || substr( $pattern, -4 ) === '.php' ) {
-				$pattern_files[] = $pattern;
+	private function get_patterns() {
+		$block_patterns = get_transient( $this->patterns_cache_key );
+
+		// Load fresh data if we don't have any patterns.
+		if ( false === $block_patterns || ( defined( 'WP_DISABLE_PATTERN_CACHE' ) && WP_DISABLE_PATTERN_CACHE ) ) {
+			$request_url = add_query_arg(
+				array(
+					'site'     => $this->source_site_id,
+					'language' => $this->get_iso_639_locale(),
+				),
+				'https://public-api.wordpress.com/rest/v1/ptk/patterns'
+			);
+
+			$response = wp_remote_get(
+				esc_url_raw( $request_url ),
+				array( 'timeout' => 20 )
+			);
+
+			if ( 200 !== wp_remote_retrieve_response_code( $response ) ) {
+				return array();
 			}
-
-			$pattern = readdir( $directory_handle );
+			$block_patterns = json_decode( wp_remote_retrieve_body( $response ), true );
+			set_transient( $this->patterns_cache_key, $block_patterns, DAY_IN_SECONDS );
 		}
 
-		closedir( $directory_handle );
-
-		// Manually curated list of patterns to go at the top of the list.
-		$featured_patterns = array(
-			'masonry-gallery.php',
-			'description-and-image.php',
-			'two-images-and-quote.php',
-			'image-and-description.php',
-			'three-images-side-by-side.php',
-			'image-and-text.php',
-			'three-columns-and-image.php',
-			'collage-gallery.php',
-			'headline.php',
-			'headline-02.php',
-			'recent-posts.php',
-			'recent-posts-02.php',
-			'two-images-side-by-side.php',
-			'numbers.php',
-			'contact.php',
-			'contact-02.php',
-			'contact-03.php',
-			'food-menu.php',
-		);
-
-		// Add the featured patterns to the top of the patterns.
-		$pattern_files = array_merge( $featured_patterns, array_diff( $pattern_files, $featured_patterns ) );
-
-		// Get the pattern files contents.
-		foreach ( $pattern_files as $pattern_file ) {
-			if ( substr( $pattern_file, -5 ) === '.json' ) {
-				$pattern_name              = self::PATTERN_NAMESPACE . substr( $pattern_file, 0, -5 );
-				$patterns[ $pattern_name ] = json_decode(
-					// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
-					file_get_contents( $patterns_dir . $pattern_file ),
-					true
-				);
-			}
+		return $block_patterns;
+	}
 
-			if ( substr( $pattern_file, -4 ) === '.php' ) {
-				$pattern_name              = self::PATTERN_NAMESPACE . substr( $pattern_file, 0, -4 );
-				$patterns[ $pattern_name ] = include $patterns_dir . $pattern_file;
-			}
+	/**
+	 * Returns ISO 639 conforming locale string.
+	 *
+	 * @return string ISO 639 locale string
+	 */
+	private function get_iso_639_locale() {
+		// Make sure to get blog locale, not user locale.
+		$language = function_exists( 'get_blog_lang_code' ) ? get_blog_lang_code() : get_locale();
+		$language = strtolower( $language );
+
+		if ( in_array( $language, array( 'pt_br', 'pt-br', 'zh_tw', 'zh-tw', 'zh_cn', 'zh-cn' ), true ) ) {
```

Comment:
I'd rather just send the language code directly and try not to duplicate lang code mapping here. @andrewserong its possibly worth having two lang fields to support both wpcom/wporg variations. (I think this will be needed but might also want to check with #i18n)

Reply:
> I think it'd be fine to leave this in for the moment and remove once we've got the additional param

> wp_locale 

lgtm*2